src/client_side_request.cc

```diff
@@ -1702,6 +1684,16 @@ ClientHttpRequest::clearRequest()
     absorbLogUri(nullptr);
 }
 
+const Ip::Address&
+ClientHttpRequest::clientAddrOnError() const
+{
+    if (const auto conn = getConn())
+        if (conn->clientConnection)
+            return conn->clientConnection->remote;
+    static Ip::Address emptyAddr;
+    return emptyAddr;
```

Comment:
We are duplicating these two lines in several places. Let's add an inlined Ip::Address::Empty() instead.